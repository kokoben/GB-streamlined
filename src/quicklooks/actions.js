export const QUICKLOOK_VIDEO_SET = 'QUICKLOOK_VIDEO_SET';
export const QUICKLOOK_VIDEO_SET_SUCCESS = 'QUICKLOOK_VIDEO_SET_SUCCEEDED';
export const QUICKLOOK_VIDEO_SET_FAIL = 'QUICKLOOK_VIDEO_SET_FAIL';
export const QUICKLOOK_VIDEOS_SET = 'QUICKLOOK_VIDEOS_SET';
export const QUICKLOOK_VIDEOS_SET_SUCCESS = 'QUICKLOOK_VIDEOS_SET_SUCCEEDED';
export const QUICKLOOK_VIDEOS_SET_FAIL = 'QUICKLOOK_VIDEOS_SET_FAIL';

export const setQuicklookVideo = () => {
  return {
    type: QUICKLOOK_VIDEO_SET
  }
}

export const setQuicklookVideos = (page, pageSize) => {
  return {
    type: QUICKLOOK_VIDEOS_SET,
    page,
    pageSize
  }
}
