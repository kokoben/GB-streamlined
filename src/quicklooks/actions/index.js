import * as qlActions from './types';

export const setQuicklookVideo = (id) => {
  console.log('inside setQuicklookVideo');
  return {
    type: qlActions.QUICKLOOK_VIDEO_SET,
    id,
  };
};

export const setQuicklookVideos = (page, pageSize) => {
  console.log('inside setQuicklookVideos');
  return {
    type: qlActions.QUICKLOOK_VIDEOS_SET,
    page,
    pageSize,
  };
};

export const setLatestQuicklookVideo = () => ({
  type: qlActions.LATEST_QUICKLOOK_VIDEO_SET,
});
