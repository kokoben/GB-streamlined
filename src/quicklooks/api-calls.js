// get latest quicklook video
export const callQuicklookVideo = 
  "https://www.giantbomb.com/api/videos/" +
  "?api_key=816627d452ffb34d20762fd2f3b575dfe906bfd9" +
  "&format=jsonp&json_callback=callback" +
  "&limit=1&field_list=name,deck,embed_player,publish_date,image," +
  "user&filter=video_type:3&sort=publish_date:desc";
