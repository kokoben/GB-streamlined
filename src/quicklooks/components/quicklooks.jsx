import React from 'react';
import QuicklookVideos from '../containers/quicklook-videos';
import CurrentQuicklookVideo from '../containers/current-quicklook-video';
import FeedHeader from '../../components/feed-header';

export default () => (
  <div>
    <CurrentQuicklookVideo />
    <FeedHeader
      category="Quicklook"
      placeholder="Search for Quick Look videos..."
    />
    <QuicklookVideos />
  </div>
);

