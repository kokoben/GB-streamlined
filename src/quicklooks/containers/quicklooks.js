import React from 'react';
import QuicklookVideos from './quicklook-videos';
import CurrentQuicklookVideo from './current-quicklook-video';

export default () => (
  <div>
    <CurrentQuicklookVideo />
    <QuicklookVideos />
  </div>
)

