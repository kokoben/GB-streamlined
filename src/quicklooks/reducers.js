import {
  QUICKLOOK_VIDEO_SET_SUCCESS,
  QUICKLOOK_VIDEO_SET_FAIL
} from './actions';

export const quicklookVideo = (state=null, action) => {
  switch(action.type) {
    case QUICKLOOK_VIDEO_SET_SUCCESS:
      console.log('quicklook_video_set_success!');
      const data = action.response.results[0];
      return data;
    case QUICKLOOK_VIDEO_SET_FAIL:
      console.log('quicklook_video_set_fail!');
      console.log(action.message);
      break;
    default:
      return state;
  }
}
