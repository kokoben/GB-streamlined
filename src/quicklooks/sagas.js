import { takeEvery, call, put } from 'redux-saga/effects';
import jsonp from 'jsonp-promise';
import {
  QUICKLOOK_VIDEO_SET,
  QUICKLOOK_VIDEO_SET_SUCCESS,
  QUICKLOOK_VIDEO_SET_FAIL
} from './actions';

const jsonpWrapper = args => {
  const response = jsonp(...args);
  return response.promise;
}

// workers
function* setQuicklookVideoAsync() {
  try {
    const jsonpArgs = [
        "https://www.giantbomb.com/api/videos/" +
        "?api_key=816627d452ffb34d20762fd2f3b575dfe906bfd9" +
        "&format=jsonp&json_callback=callback" +
        "&limit=1&field_list=name,deck,embed_player,publish_date,image," +
        "user&filter=video_type:3&sort=publish_date:desc",
        {param: 'json_callback'}
    ];

    const response = yield call(jsonpWrapper, jsonpArgs);
    yield put({type: QUICKLOOK_VIDEO_SET_SUCCESS, response: response});
  } catch (e) {
    console.log('quicklook_video_set_failed!');
    yield put({type: QUICKLOOK_VIDEO_SET_FAIL, message: e.message});
  }

}

// watchers
export function* watchSetQuicklookVideo() {
  console.log('redux-saga is running the QUICKLOOK_VIDEO_SET action listener');
  yield takeEvery(QUICKLOOK_VIDEO_SET, setQuicklookVideoAsync);
}
