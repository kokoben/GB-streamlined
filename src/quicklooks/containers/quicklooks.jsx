import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import QuicklookVideos from '../containers/quicklook-videos';
import CurrentQuicklookVideo from '../containers/current-quicklook-video';
import QuicklookFeedHeader from '../components/quicklook-feedheader';
import { setSearchPage } from '../../actions/index';

export class Quicklooks extends Component {
  componentDidMount() {
    this.props.setSearchPage(null);
  }

  render() {
    return (
      <div>
        <CurrentQuicklookVideo />
        <QuicklookFeedHeader />
        <QuicklookVideos />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setSearchPage,
  }, dispatch)
);

Quicklooks.propTypes = {
  setSearchPage: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(Quicklooks);
