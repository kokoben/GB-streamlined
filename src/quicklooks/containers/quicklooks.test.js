import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Quicklooks } from './quicklooks';

it('renders correctly', () => {
  const mockSetSearchPage = jest.fn();
  const wrapper = shallow(<Quicklooks
    setSearchPage={mockSetSearchPage}
  />);

  const tree=toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
