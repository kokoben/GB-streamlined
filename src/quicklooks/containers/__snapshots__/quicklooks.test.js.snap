// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <Connect(CurrentQuicklookVideo) />
  <Connect(Component) />
  <Connect(QuicklookVideos) />
</div>
`;
